app/models/topic.rb

```diff
@@ -655,16 +654,17 @@ def add_moderator_post(user, text, opts=nil)
                               skip_validations: true,
                               custom_fields: opts[:custom_fields])
 
-    if (new_post = creator.create) && new_post.present?
-      increment!(:moderator_posts_count) if new_post.persisted?
-      # If we are moving posts, we want to insert the moderator post where the previous posts were
-      # in the stream, not at the end.
-      new_post.update_attributes!(post_number: opts[:post_number], sort_order: opts[:post_number]) if opts[:post_number].present?
+    new_post = creator.create
 
-      # Grab any links that are present
-      TopicLink.extract_from(new_post)
-      QuotedPost.extract_from(new_post)
-    end
+    increment!(:moderator_posts_count) if new_post.persisted?
+
+    # If we are moving posts, we want to insert the moderator post where the previous posts were
+    # in the stream, not at the end.
+    new_post.update_columns(post_number: opts[:post_number], sort_order: opts[:post_number]) if opts[:post_number].present?
```

Comment:
@SamSaffron Do you mean the fact of updating the post number here or the fact that I'm removing the validation?

If it's the latter, I removed the validation because it was getting in the middle for us. Validator posts shouldn't follow the same validation rules as other posts. For example, they shouldn't have a minimum length, which was making moving some posts crash for us. But you're totally right about post number uniqueness, not sure if that could happen, but it's definitely something to be aware of.